task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -392,36 +392,47 @@ def _pending_event_processing_loop(self) -> None:
 
         logger.debug("Exiting Pod event processing - stop requested.")
 
+    def _create_container_definition(
+        self,
+        name: str,
+        task_config: KubernetesTaskConfig,
+    ) -> V1Container:
+        return V1Container(
+            image=task_config.image,
+            name=name,
+            command=["/bin/sh", "-c"],
+            args=[task_config.command],
+            security_context=get_security_context_for_capabilities(
+                cap_add=task_config.cap_add,
+                cap_drop=task_config.cap_drop,
+            ),
+            resources=V1ResourceRequirements(
+                limits={
+                    "cpu": task_config.cpus,
+                    "memory": f"{task_config.memory}Mi",
+                    "ephemeral-storage": f"{task_config.disk}Mi",
+                }
+            ),
+            env=get_kubernetes_env_vars(
+                environment=task_config.environment,
+                secret_environment=task_config.secret_environment,
+                field_selector_environment=task_config.field_selector_environment,
+            ),
+            volume_mounts=get_kubernetes_volume_mounts(task_config.volumes),
+            ports=[V1ContainerPort(container_port=port) for port in task_config.ports],
```

Comment:
hmm, do we need to validate against multiple containers trying to expose the same port?

Reply:
Oh, is that not allowed

Reply:
Oh, is that something that is not allowed?

Reply:
i'm...not sure - i can take a quick look, was mostly just thinking out loud

Reply:
yea, afaict from reading the docs: ports in a pod need to be unique